src/Components/Nav/Nav.js

```diff
@@ -0,0 +1,317 @@
+import React, { useState, useEffect, Fragment } from 'react';
+import { Link, withRouter, useLocation } from 'react-router-dom';
+import styled from 'styled-components/macro';
+import ClickedSearchBar from './Components/ClickedSearchBar/ClickedSearchBar';
+import { Search } from '@styled-icons/fa-solid';
+import { ArrowDown } from '@styled-icons/evaicons-solid';
+import { API } from '../../../src/config';
+
+function Nav(props) {
+  const [categoryDatas, setCategoryDatas] = useState();
+  const [onMouseBigCategory, setOnMouseBigCategory] = useState(false);
+  const [onMouseSubCategory, setOnMouseSubCategory] = useState(false);
+  const [selectedCategory, setSelectedCategory] = useState();
+  const [clickedSearchBar, setClickedSearchBar] = useState(false);
+  const [searchLog, setSearchLog] = useState(
+    JSON.parse(localStorage.getItem('search log')) || []
+  );
+  const [login, setLogin] = useState('로그인');
+  const location = useLocation().pathname;
+
+  useEffect(() => {
+    fetch(`${API}/courses/category`)
+      .then(res => res.json())
+      .then(data => setCategoryDatas(data.category));
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('search log', JSON.stringify(searchLog));
+  }, [searchLog]);
+
+  useEffect(() => {
+    if (localStorage.getItem('kakao_4bc6915671ecc50762f3c678ed51a503')) {
+      setLogin('로그아웃');
+    } else {
+      setLogin('로그인');
+    }
+  }, [login]);
+
+  const handleLogin = () => {
+    if (login === '로그인') {
+      goToLoginPage();
+    } else {
+      localStorage.removeItem('kakao_4bc6915671ecc50762f3c678ed51a503');
+      setLogin('로그인');
+    }
+  };
+
+  const addLog = e => {
+    setSearchLog([...searchLog, { id: Date.now(), word: e.target[0].value }]);
+    e.target[0].value = '';
+  };
+  const delLog = log => {
+    setSearchLog(searchLog.filter(logToDel => logToDel.id !== log.id));
+  };
+
+  const goToMainPage = e => {
+    props.history.push(`/`);
+  };
+
+  const goToCategoryPage = e => {
+    props.history.push(`/category/${e.target.textContent}`);
+  };
+
+  const goToCreateClassPage = () => {
+    props.history.push(`/create`);
+  };
+
+  const goToMyPage = () => {
+    props.history.push(`/mypage`);
+  };
+
+  const goToLoginPage = () => {
+    props.history.push(`/login`);
+  };
```

Comment:
```suggestion
  const goToPage = e => {
  // 각 영역에 name 값을 넣어주고, ex) create, mypage 등...
    props.history.push(`/${e.target.name}`);
  };
```
이런식으로 하나의 함수로 합쳐볼까요?
category의 경우만 조건 걸어서 빼주면 되겠네요~

Reply:
ㅎㅎㅎㅎ 완전히 비효율적인 코드였네요 ㅎㅎ 하나로 합쳤습니당!